### PR TITLE
Fixed the reported issue regarding #393

### PR DESCRIPTION
**Pull request by BNMetrics**
This PR is related to the ticket [393](https://github.com/thought-machine/please/issues/393)
- Added absolute path in coverage.xml generation, as intellij only recognize absolute path. 
- Added relative path to class names, same reason as above

### CODE REVIEW

File: src/test/xml_coverage.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ import (
 	"strings"
 	"time"
 	"math"
+		"os"
```
Comment (peterebden) [inline src/test/xml_coverage.go]: please run `gofmt` on this file (and all Go files)

File: src/test/xml_coverage.go [hunk 2]
```diff
@@ -48,16 +49,22 @@ func coverageResultToXML(sources []core.BuildLabel, coverage core.TestCoverage)
 	linesCovered := 0
 	validFiles := coverage.OrderedFiles()
 
+	cwd, err := os.Getwd()
```
Comment (peterebden) [inline src/test/xml_coverage.go]: I don't think you should find the working directory here. Use `core.RepoRoot` if you need to know it - but I don't think you should be encoding absolute paths into the output file.

Comment (BNMetrics) [reply src/test/xml_coverage.go]: I just tried it out, using relative path would show 0 covered in intellij. :(

Comment (peterebden) [reply src/test/xml_coverage.go]: Intellij is terrible :(

I'm not sure what is best then. These files can get shared through the cache, having absolute paths in them is not a good idea.

Comment (agenticarus) [reply src/test/xml_coverage.go]: Perhaps you could write them with relative paths and have a separate post-process step that outputs the file for intellij that doesn't get cached (this is likely to help if we ever want to support other IDEs or formats too)?

Comment (peterebden) [reply src/test/xml_coverage.go]: Hmmm on further thought the output file is *not* cached so I guess that's OK.

I still think it's rubbish that Intellij doesn't handle relative paths though.

File: src/test/xml_coverage.go [hunk 3]
```diff
@@ -51,13 +51,13 @@ func coverageResultToXML(sources []core.BuildLabel, coverage core.TestCoverage)
 	// get the string representative of sources
 	var sourcesAsStr []string
 	for _, source := range sources {
-		sourcesAsStr = append(sourcesAsStr, source.String())
+		sourcesAsStr = append(sourcesAsStr, core.RepoRoot+"/"+source.PackageName)
```
Comment (peterebden) [inline src/test/xml_coverage.go]: use `path.Join` to join paths rather than concatenating.

File: src/test/xml_coverage.go [hunk 4]
```diff
@@ -51,13 +52,13 @@ func coverageResultToXML(sources []core.BuildLabel, coverage core.TestCoverage)
 	// get the string representative of sources
 	var sourcesAsStr []string
 	for _, source := range sources {
-		sourcesAsStr = append(sourcesAsStr, source.String())
+		sourcesAsStr = append(sourcesAsStr, path.Join(core.RepoRoot, "/", source.PackageName))
```
Comment (peterebden) [inline src/test/xml_coverage.go]: you don't need the "/" if you're using JoinPath - let it add it for you if necessary.

**Pull request merged**